llvm-external-projects/iree-dialects/python/mlir/dialects/iree_pydm/rtl/modules/macros.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2021 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Macros that can be used freely when building RTL modules."""
+
+from ...importer import (
+    def_ir_macro_intrinsic,
+    ImportStage,
+)
+
+from .... import iree_pydm as d
+from ..... import ir
```

Comment:
another TIL - but not sure I wanted to know this one ;P

Reply:
Yeah - in MLIR dialects we keep them location independent, which... stretches credulity (but is better than hacks to try to do the same). You just keep adding dots until IntelliSense lights up.